### PR TITLE
chore: Refactoring 

### PR DESCRIPTION
**Pull request by samkahchiin**
#### Summary
- cloudflare: 
  - Allow users to declare the `s3_cloudflare_records` instead of hardcording it to [cdn, app] 
  - Allow user to declare the worker script name
- elasticache:
  - fix cluster mode elasticache endpoint output
- stack/app:
  - Add `alb_dns_name` as output
  - Make `db_instance_class_memory_in_gb` as optional variables
- stack/app
  - Add cloudflare TLS setup

### CODE REVIEW

File: stack/setup/README.md [hunk 1]
```diff
@@ -59,6 +59,17 @@ module "stack-setup" {
...
 }
+
+# versions.tf
+
+# export AWS_ACCESS_KEY_ID=
+# export AWS_SECRET_ACCESS_KEY=
+# export AWS_REGION=
+provider "aws" {}
+
+# export CLOUDFLARE_EMAIL=
```
Comment (swiknaba) [inline stack/setup/README.md:70]: Didn't you remove email elsewhere?

File: stack/setup/cloudflare.tf [hunk 2]
```diff
@@ -33,3 +33,18 @@ resource "aws_acm_certificate_validation" "default" {
   certificate_arn         = aws_acm_certificate.main.arn
   validation_record_fqdns = cloudflare_record.validation.*.hostname
 }
+
+# If you came across this error:
+# Error: invalid zone setting "cname_flattening" (value: ) found - cannot be set as it is read only
+# 1. terraform state rm module.<module-name>.cloudflare_zone_settings_override.tls
+# 2. terraform apply
+# https://github.com/cloudflare/terraform-provider-cloudflare/issues/808
+resource "cloudflare_zone_settings_override" "tls" {
```
Comment (swiknaba) [inline stack/setup/cloudflare.tf:42]: Should we move this to the Cloudflare module? We can pass the settings as variable, with a default as you have it here. If null is passed explicitly, then we skip this. WDYT?

File: stack/setup/cloudflare.tf [hunk 3]
```diff
@@ -33,3 +33,18 @@ resource "aws_acm_certificate_validation" "default" {
...
+  zone_id = data.cloudflare_zone.default.id
+
+  settings {
```
Comment (swiknaba) [inline stack/setup/cloudflare.tf:45]: should we additionally set `always_use_https` to "on"? (defaults to "off").

Always Use HTTPS
> Redirect all requests with scheme “http” to “https”. This applies to all http requests to the zone.

Comment (samkahchiin) [reply stack/setup/cloudflare.tf]: Good idea! have added this to settings

**Review by swiknaba — APPROVED** [review]
awesome, thanks :)

**Pull request merged**